Best Practices/Challenges/DI/DI.Challenge/DIChallenge.cs

```diff
@@ -0,0 +1,25 @@
+﻿using DI.Challenge.Services;
+
+namespace DI.Challenge
+{
+    public class DIChallenge
+    {
+        private readonly TransientService _transientService;
+        private readonly ScopedService _scopedService;
+        private readonly SingletonService _singletonService;
+
+        public DIChallenge()
+        {
+            _transientService = new TransientService();
+            _scopedService = new ScopedService();
+            _singletonService = new SingletonService();
```

Comment:
Suggestion: Could we do a skill level build up where -

1. The first test has a class that depends on a singleton concrete
2. The second test has a class that depends on a singleton interface
3. This test where we ask for 3 multiple lifetimes
4. The fourth depends on multiple implementations of an interface (IEnumerable<T>). We could give them two classes that implement an interface.
5. A factory registration - not sure how this would look though, maybe skip.
6. Keyed registration